File: TestProject1/Module/UnitTest_MdlApp.cs
using System.Diagnostics;
using System.Reflection;
using System.Runtime.InteropServices;
using CmnClsLib.Module;
using Xunit;
using Assert = Xunit.Assert;

namespace TestProject1.Module
{
    public class UnitTest_MdlApp
    {
        // --------------------------------------------------------------------
        // IsWindows()
        // --------------------------------------------------------------------
        [Fact]
        public void IsWindows_OS環境の判定結果を返すこと()
        {
            bool expected = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
            bool actual = MdlApp.IsWindows();
            Assert.Equal(expected, actual);
        }

        // --------------------------------------------------------------------
        // GetOsName()
        // --------------------------------------------------------------------
        [Fact]
        public void GetOsName_OS名を返すこと()
        {
            string actual = MdlApp.GetOsName();
            Assert.False(string.IsNullOrEmpty(actual));
            Assert.Contains(actual, new[] { "Windows", "Linux", "OSX", "FreeBSD", "UNKNOWN" });
        }

        // --------------------------------------------------------------------
        // GetProcessArchitecture()
        // --------------------------------------------------------------------
        [Fact]
        public void GetProcessArchitecture_プロセスのアーキテクチャを返すこと()
        {
            string actual = MdlApp.GetProcessArchitecture();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // GetOsArchitecture()
        // --------------------------------------------------------------------
        [Fact]
        public void GetOsArchitecture_OSのアーキテクチャを返すこと()
        {
            string actual = MdlApp.GetOsArchitecture();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // GetOsDescription()
        // --------------------------------------------------------------------
        [Fact]
        public void GetOsDescription_OSの詳細説明文字列を返すこと()
        {
            string actual = MdlApp.GetOsDescription();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // GetOsVersion()
        // --------------------------------------------------------------------
        [Fact]
        public void GetOsVersion_OSのバージョン文字列を返すこと()
        {
            string actual = MdlApp.GetOsVersion();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // GetOsMajorVersion()
        // --------------------------------------------------------------------
        [Fact]
        public void GetOsMajorVersion_OSのメジャーバージョンを返すこと()
        {
            int actual = MdlApp.GetOsMajorVersion();
            Assert.True(actual > 0);
        }

        // --------------------------------------------------------------------
        // GetPlatform()
        // --------------------------------------------------------------------
        [Fact]
        public void GetPlatform_プラットフォーム識別子を返すこと()
        {
            string actual = MdlApp.GetPlatform();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // GetReferencedAssembliesDictionary() / GetReferencedAssembliesDict()
        // --------------------------------------------------------------------
        [Fact]
        public void GetReferencedAssembliesDictionary_Nullアセンブリの場合は空の辞書を返すこと()
        {
            var actual = MdlApp.GetReferencedAssembliesDictionary(null!, "ALL");
            Assert.Empty(actual);
        }

        [Fact]
        public void GetReferencedAssembliesDictionary_ALL指定で参照アセンブリを取得できること()
        {
            Assembly asm = Assembly.GetExecutingAssembly();
            var actual = MdlApp.GetReferencedAssembliesDictionary(asm, "ALL");
            Assert.NotEmpty(actual);
        }

        [Fact]
        public void GetReferencedAssembliesDictionary_Nullまたは空文字キー指定でフィルタリング処理されること()
        {
            Assembly asm = Assembly.GetExecutingAssembly();
            var actual = MdlApp.GetReferencedAssembliesDictionary(asm, "");
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetReferencedAssembliesDictionary_特定キー指定でフィルタリング処理されること()
        {
            Assembly asm = Assembly.GetExecutingAssembly();
            var actual = MdlApp.GetReferencedAssembliesDictionary(asm, "b77a5c561934e089");
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetReferencedAssembliesDict_非推奨メソッドが同等の結果を返すこと()
        {
            Assembly asm = Assembly.GetExecutingAssembly();
            var expected = MdlApp.GetReferencedAssembliesDictionary(asm, "ALL");
#pragma warning disable CS0618
            var actual = MdlApp.GetReferencedAssembliesDict(asm, "ALL");
#pragma warning restore CS0618
            Assert.Equal(expected.Count, actual.Count);
        }

        // --------------------------------------------------------------------
        // ランタイム / CLR 情報
        // --------------------------------------------------------------------
        [Fact]
        public void GetFrameworkDescription_フレームワークの説明文字列を返すこと()
        {
            string actual = MdlApp.GetFrameworkDescription();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        [Fact]
        public void GetRuntimeClrVersion_ランタイムCLRバージョンを返すこと()
        {
            string actual = MdlApp.GetRuntimeClrVersion();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        [Fact]
        public void GetBuildClrVersion_ビルド時CLRバージョンを返すこと()
        {
            string actual = MdlApp.GetBuildClrVersion();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        // --------------------------------------------------------------------
        // プロセス / アセンブリ 情報
        // --------------------------------------------------------------------
        [Fact]
        public void GetExeFilePath_実行ファイルのパスを返すこと()
        {
            string actual = MdlApp.GetExeFilePath();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetExeFileVersion_実行ファイルのバージョンを返すこと()
        {
            string actual = MdlApp.GetExeFileVersion();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetExeName_実行ファイル名を返すこと()
        {
            string actual = MdlApp.GetExeName();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetAssemblyLocation_アセンブリのロケーションを返すこと()
        {
            string actual = MdlApp.GetAssemblyLocation();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetAssemblyName_アセンブリ名を返すこと()
        {
            string actual = MdlApp.GetAssemblyName();
            Assert.False(string.IsNullOrEmpty(actual));
        }

        [Fact]
        public void GetAssemblyVersion_アセンブリバージョンを返すこと()
        {
            string actual = MdlApp.GetAssemblyVersion();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetAppName_アプリケーション名を返すこと()
        {
            string actual = MdlApp.GetAppName();
            Assert.NotNull(actual);
        }

        [Fact]
        public void GetAppNameWithHostName_アプリ名と小文字のホスト名を結合した文字列を返すこと()
        {
            string actual = MdlApp.GetAppNameWithHostName();
            string expectedAppName = MdlApp.GetAppName();
            string expectedHostName = Environment.MachineName.ToLower();
            Assert.Equal($"{expectedAppName}_{expectedHostName}", actual);
        }

        [Fact]
        public void GetAppName_HostName_非推奨メソッドが同等の結果を返すこと()
        {
#pragma warning disable CS0618
            string actual = MdlApp.GetAppName_HostName();
#pragma warning restore CS0618
            Assert.Equal(MdlApp.GetAppNameWithHostName(), actual);
        }

        // --------------------------------------------------------------------
        // ProcessPriorityClass 関連
        // --------------------------------------------------------------------
        [Fact]
        public void GetPriorityName_優先度クラス名文字列を正しく変換すること()
        {
            Assert.Equal("BelowNormal", MdlApp.GetPriorityName(ProcessPriorityClass.BelowNormal));
            Assert.Equal("High", MdlApp.GetPriorityName(ProcessPriorityClass.High));
            Assert.Equal("Idle", MdlApp.GetPriorityName(ProcessPriorityClass.Idle));
            Assert.Equal("Normal", MdlApp.GetPriorityName(ProcessPriorityClass.Normal));
            Assert.Equal("RealTime", MdlApp.GetPriorityName(ProcessPriorityClass.RealTime));
            Assert.Equal("AboveNormal", MdlApp.GetPriorityName(ProcessPriorityClass.AboveNormal));
            Assert.Equal("AboveNormal", MdlApp.GetPriorityName((ProcessPriorityClass)999));
        }

        [Fact]
        public void GetPriorityClassFromString_文字列から優先度クラスへ正しく変換すること()
        {
            Assert.Equal(ProcessPriorityClass.Normal, MdlApp.GetPriorityClassFromString(null!));
            Assert.Equal(ProcessPriorityClass.Normal, MdlApp.GetPriorityClassFromString(""));
            Assert.Equal(ProcessPriorityClass.AboveNormal, MdlApp.GetPriorityClassFromString("abovenormal"));
            Assert.Equal(ProcessPriorityClass.AboveNormal, MdlApp.GetPriorityClassFromString("ABOVENORMAL"));
            Assert.Equal(ProcessPriorityClass.Idle, MdlApp.GetPriorityClassFromString("idle"));
            Assert.Equal(ProcessPriorityClass.High, MdlApp.GetPriorityClassFromString("high"));
            Assert.Equal(ProcessPriorityClass.RealTime, MdlApp.GetPriorityClassFromString("realtime"));
            Assert.Equal(ProcessPriorityClass.BelowNormal, MdlApp.GetPriorityClassFromString("belownormal"));
            Assert.Equal(ProcessPriorityClass.Normal, MdlApp.GetPriorityClassFromString("invalid_value"));
        }

        [Fact]
        public void GetPriorityClassFromStr_非推奨メソッドが同等の結果を返すこと()
        {
#pragma warning disable CS0618
            Assert.Equal(MdlApp.GetPriorityClassFromString("high"), MdlApp.GetPriorityClassFromStr("high"));
#pragma warning restore CS0618
        }
    }
}

File: src/test/java/tool/cmnclslib/mdl/UnitTest_MdlApp.java
package tool.cmnclslib.mdl;

import org.junit.Test;
import static org.junit.Assert.*;

/**
 * MdlApp の単体テストクラスです。
 */
public class UnitTest_MdlApp {

    @Test
    public void isWindows_OS環境の判定結果を返すこと() {
        boolean actual = MdlApp.isWindows();
        String os = System.getProperty("os.name").toLowerCase();
        boolean expected = os.contains("win");
        assertEquals(expected, actual);
    }

    @Test
    public void getOsName_OS名を返すこと() {
        String actual = MdlApp.getOsName();
        assertFalse(actual.isEmpty());
        assertTrue(actual.equals("Windows") || actual.equals("Linux") || actual.equals("OSX") || actual.equals("FreeBSD") || actual.equals("UNKNOWN"));
    }

    @Test
    public void getProcessArchitecture_プロセスのアーキテクチャを返すこと() {
        String actual = MdlApp.getProcessArchitecture();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getOsArchitecture_OSのアーキテクチャを返すこと() {
        String actual = MdlApp.getOsArchitecture();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getOsDescription_OSの詳細説明文字列を返すこと() {
        String actual = MdlApp.getOsDescription();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getOsVersion_OSのバージョン文字列を返すこと() {
        String actual = MdlApp.getOsVersion();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getOsMajorVersion_OSのメジャーバージョンを返すこと() {
        int actual = MdlApp.getOsMajorVersion();
        assertTrue(actual > 0);
    }

    @Test
    public void getPlatform_プラットフォーム識別子を返すこと() {
        String actual = MdlApp.getPlatform();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getFrameworkDescription_フレームワークの説明文字列を返すこと() {
        String actual = MdlApp.getFrameworkDescription();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getRuntimeClrVersion_ランタイムバージョンを返すこと() {
        String actual = MdlApp.getRuntimeClrVersion();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getBuildClrVersion_ビルドバージョンを返すこと() {
        String actual = MdlApp.getBuildClrVersion();
        assertFalse(actual.isEmpty());
    }

    @Test
    public void getExeFilePath_実行ファイルのパスを返すこと() {
        String actual = MdlApp.getExeFilePath();
        assertNotNull(actual);
    }

    @Test
    public void getExeFileVersion_実行ファイルのバージョンを返すこと() {
        String actual = MdlApp.getExeFileVersion();
        assertNotNull(actual);
    }

    @Test
    public void getExeName_実行ファイル名を返すこと() {
        String actual = MdlApp.getExeName();
        assertNotNull(actual);
    }

    @Test
    public void getAppName_アプリケーション名を返すこと() {
        String actual = MdlApp.getAppName();
        assertNotNull(actual);
    }

    @Test
    public void getAppNameWithHostName_アプリ名と小文字のホスト名を結合した文字列を返すこと() {
        String actual = MdlApp.getAppNameWithHostName();
        String expectedAppName = MdlApp.getAppName();
        String host = "";
        try {
            host = java.net.InetAddress.getLocalHost().getHostName().toLowerCase(java.util.Locale.ROOT);
        } catch (Exception e) {
            host = "localhost";
        }
        assertEquals(expectedAppName + "_" + host, actual);
    }

    @Test
    public void getPriorityName_優先度クラス名文字列を正しく変換すること() {
        assertEquals("BelowNormal", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.BELOW_NORMAL));
        assertEquals("High", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.HIGH));
        assertEquals("Idle", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.IDLE));
        assertEquals("Normal", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.NORMAL));
        assertEquals("RealTime", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.REAL_TIME));
        assertEquals("AboveNormal", MdlApp.getPriorityName(MdlApp.ProcessPriorityClass.ABOVE_NORMAL));
    }

    @Test
    public void getPriorityClassFromString_文字列から優先度クラスへ正しく変換すること() {
        assertEquals(MdlApp.ProcessPriorityClass.NORMAL, MdlApp.getPriorityClassFromString(null));
        assertEquals(MdlApp.ProcessPriorityClass.NORMAL, MdlApp.getPriorityClassFromString(""));
        assertEquals(MdlApp.ProcessPriorityClass.ABOVE_NORMAL, MdlApp.getPriorityClassFromString("abovenormal"));
        assertEquals(MdlApp.ProcessPriorityClass.ABOVE_NORMAL, MdlApp.getPriorityClassFromString("ABOVENORMAL"));
        assertEquals(MdlApp.ProcessPriorityClass.IDLE, MdlApp.getPriorityClassFromString("idle"));
        assertEquals(MdlApp.ProcessPriorityClass.HIGH, MdlApp.getPriorityClassFromString("high"));
        assertEquals(MdlApp.ProcessPriorityClass.REAL_TIME, MdlApp.getPriorityClassFromString("realtime"));
        assertEquals(MdlApp.ProcessPriorityClass.BELOW_NORMAL, MdlApp.getPriorityClassFromString("belownormal"));
        assertEquals(MdlApp.ProcessPriorityClass.NORMAL, MdlApp.getPriorityClassFromString("invalid_value"));
    }

    @Test
    public void 新設メソッド群が正しく動作すること() {
        assertNotNull(MdlApp.getJavaVersion());
        assertNotNull(MdlApp.getProcessArch());
        assertNotNull(MdlApp.getFrameworkDesc());
        assertNotNull(MdlApp.getAppNameWithHost());
        assertNotNull(MdlApp.getRefPackagesMap(UnitTest_MdlApp.class, ""));
        assertEquals(MdlApp.ProcessPriorityClass.HIGH, MdlApp.parsePriorityClass("high"));
    }
}
